### DISCUSSION TITLE
Image variations

### DISCUSSION THREAD

**Discussion by morganavr** [hlky/stable-diffusion-webui#595] [Feature suggestions]
It's a very powerful and addicting feature.

Example of generated images from "variations" function:

![Text](https://user-images.githubusercontent.com/86247933/187781637-ae95f234-1330-4b6d-8f02-02e06b49573e.gif)

This feature allows to control variations amount via `-v` parameter allowing to get infinite variations of a generated image.
First, you generate an image and remember it's Seed.
Then you can create variation of the original image with this prompt: `"chinese girl" -S 777777777 -v 0.005`
0.005 means that you will get the image that will be different on 0.5% compared to original. Important thing to understand is that variations work great with txt2img module, no need to provide initial image like in img2img module.

This functionality is already here: https://github.com/lstein/stable-diffusion/pull/277
But it requires complex prompts like
`"chinese girl" -S3357757885 -V3647897225,0.1;1614299449,0.02;-V3647347225,0.15;;-V364756225,0.3;;-V32447897225,0.0055;;-V3647897225,0.003;;-V3647897225,0.0038;;-V3647897225,0.0018; -v0.05 -n6`

Originally it was implemented at https://github.com/lstein/stable-diffusion/pull/184 but it had a problem that variations of variations of variations images could not be created but I fixed it with this code: https://github.com/lstein/stable-diffusion/pull/184#issuecomment-1233426883 and you can just type ``"chinese girl" -S 777777777 -v 0.005`` to create variations of variations. Short prompt is possible because I save tensor data in files placed next to images.

Variations can be improved even more if combined with masking (to freeze parts of the variation image you don't want to be changed). Discussion of adding a masking feature to variation is here: https://github.com/lstein/stable-diffusion/issues/269

**Comment by trufty** [comment]
This is already implemented though?  
https://github.com/hlky/stable-diffusion-webui/blob/e5e16634577fa68bba3fe83a9bc4547e92683232/frontend/frontend.py#L99
Check under advanced tab on txt2image

**Comment by EliEron** [comment]
This has already been implemented in PR #307, and it's behavior around batch operations with variant seeds was improved in PR #574.

**Comment by morganavr** [comment]
Sorry, I did not see variations in a list of features in README.md that's why I opened this discussion.
